webapp/channels/src/components/admin_console/system_roles/system_roles.tsx

```diff
@@ -45,69 +45,70 @@ const columns: Column[] = [
     },
 ];
 
-export default class SystemRoles extends React.PureComponent<Props> {
-    render() {
-        const {roles} = this.props;
-        const roleNames = ['system_admin', 'system_manager', 'system_user_manager', 'system_custom_group_admin', 'system_read_only_admin'];
-        const rows: Row[] = [];
-        roleNames.forEach((name) => {
-            const role = roles[name];
-            if (role) {
-                rows.push({
-                    cells: {
-                        role: <FormattedMessage {...rolesStrings[role.name].name}/>,
-                        description: <FormattedMessage {...rolesStrings[role.name].description}/>,
-                        type: <FormattedMessage {...rolesStrings[role.name].type}/>,
-                        edit: (
-                            <span
-                                className='SystemRoles_editRow'
-                                data-testid={`${role.name}_edit`}
-                            >
-                                <Link to={`/admin_console/user_management/system_roles/${role.id}`} >
-                                    <FormattedMessage
-                                        id='admin.permissions.roles.edit'
-                                        defaultMessage='Edit'
-                                    />
-                                </Link>
-                            </span>
-                        ),
-                    },
-                    onClick: () => getHistory().push(`/admin_console/user_management/system_roles/${role.id}`),
-                });
-            }
-        });
+const roleNames = ['system_admin', 'system_manager', 'system_user_manager', 'system_custom_group_admin', 'system_read_only_admin'];
 
-        return (
-            <div className='wrapper--fixed'>
-                <AdminHeader>
-                    <FormattedMessage
-                        id='admin.permissions.systemRoles'
-                        defaultMessage='Delegated Granular Administration'
-                    />
-                </AdminHeader>
-                <div className='admin-console__wrapper'>
-                    <div className='admin-console__content'>
-                        <AdminPanel
-                            id='SystemRoles'
-                            title={defineMessage({id: 'admin.permissions.systemRolesBannerTitle', defaultMessage: 'Admin Roles'})}
-                            subtitle={defineMessage({id: 'admin.permissions.systemRolesBannerText', defaultMessage: 'Manage different levels of access to the system console.'})}
+const SystemRoles = ({roles}: Props) => {
+    const rows: Row[] = [];
```

Comment:
Use a `useMemo` to store rows. If not, you are iterating through the array on every render, and creating a new rows list on every render that is being passed as a prop to DataGrid.